enterprise_access/apps/content_assignments/tasks.py

```diff
@@ -12,17 +12,142 @@
 from enterprise_access.apps.api_client.braze_client import BrazeApiClient
 from enterprise_access.apps.api_client.lms_client import LmsApiClient
 from enterprise_access.apps.content_assignments.constants import AssignmentActionErrors, AssignmentActions
-from enterprise_access.apps.content_assignments.content_metadata_api import get_content_metadata_for_assignments
+from enterprise_access.apps.content_assignments.content_metadata_api import (
+    get_card_image_url,
+    get_content_metadata_for_assignments,
+    get_course_partners,
+    get_human_readable_date
+)
 from enterprise_access.apps.content_assignments.models import LearnerContentAssignmentAction
 from enterprise_access.tasks import LoggedTaskWithRetry
 
 from .constants import LearnerContentAssignmentStateChoices
-from .exceptions import MissingContentAssignment, MissingPolicy
+from .exceptions import MissingContentAssignment
 from .utils import format_traceback
 
 logger = logging.getLogger(__name__)
 
 
+class BrazeCampaignSender:
+    """
+    Class to help standardize the allowed keys and methods of conversion to values
+    for Braze API-triggered campaign properties.  Use as follows:
+
+    sender = BrazeCampaignSender(learner_content_assignment_record)
+    props = sender.get_properties(course_title, course_partner, ...) # any subset of ALLOWED_TRIGGER_PROPERTIES
+    sender.send_campaign_message(props, campaign_identifier)
+    """
+    ALLOWED_TRIGGER_PROPERTIES = {
+        'contact_admin_link',
+        'organization',
+        'course_title',
+        'enrollment_deadline',
+        'start_date',
+        'course_partner',
+        'course_card_image',
+        'learner_portal_link'
+    }
+
+    def __init__(self, assignment):
+        self.assignment = assignment
+        self.enterprise_customer_uuid = assignment.assignment_configuration.enterprise_customer_uuid
+
+        subsidy_policy_model = apps.get_model('subsidy_access_policy.SubsidyAccessPolicy')
+        try:
+            self.policy = subsidy_policy_model.objects.get(
+                assignment_configuration=assignment.assignment_configuration
+            )
+        except subsidy_policy_model.DoesNotExist:
+            logger.warning(f'policy with assignment config: {assignment.assignment_configuration} does not exist.')
+            raise
+
+        self.braze_client = BrazeApiClient()
+        self.lms_client = LmsApiClient()
+        self._customer_data = None
+        self._course_metadata = None
+
+    def send_campaign_message(self, braze_trigger_properties, campaign_identifier):
+        """
+        Creates a recipient and sends a braze campaign message.
+        """
+        recipient = self.braze_client.create_recipient(
+            user_email=self.assignment.learner_email,
+            lms_user_id=self.assignment.lms_user_id,
+        )
```

Comment:
@iloveagent57 Do we need to check if the lms_user_id has been created in Braze before triggering the campaign?